src/api/app/controllers/webui/project_controller.rb

```diff
@@ -93,9 +93,31 @@ def subprojects
   end
 
   def pulse
+    switch_to_webui2
     @pulse = @project.project_log_entries.page(params[:page])
   end
 
+  def update_pulse
+    sleep(2)
+    @range = 'month' if params[:range] == 'month'
+    @range ||= 'week'
+    case @range
+    when 'month'
+      range = 1.month.ago..Date.tomorrow
+    else
+      range = 1.week.ago..Date.tomorrow
+    end
+
+    @pulse = @project.project_log_entries.where(datetime: range).group(:event_type).count
+
+    @requests = @project.target_of_bs_requests.where(updated_at: range).group(:state).count
+    # sort by value...
+    @requests = @requests.sort_by(&:last).reverse.to_h
+    # transpose to percentages
+    @requests_by_percentage = @requests.each_with_object({}) { |(k, v), hash| hash[k] = "#{(v * 100.0 / @requests.values.sum).round}" } if @requests.any?
+    switch_to_webui2
+  end
```

Comment:
Looks like this deserves a controller of it's own. Please create one:-)

Reply:
I would like to do this in a later PR if you don't mind...

Reply:
But why? The existing action was a one-liner so there is not much to move.

Reply:
time...

Reply:
Took about 30 minutes:wink:
https://github.com/hennevogel/open-build-service/pull/8/commits/a8b6a400049b5c43f026f00086b004cd155a16c3